### ISSUE TITLE
Could not compile `pear_codegen`

### ISSUE THREAD

**Issue by nabeelomer** [SergioBenitez/Rocket#507]
Building a Rocket based project fails with the following error. Cargo.lock shows `pear_codegen` is required by Rocket.

Rocket v3.3.4
MacOS 10.13.2
rustc 1.24.0-nightly (dc39c3169 2017-12-17)

```
...
error[E0599]: no method named `unwrap` found for type `syntax::ptr::P<syntax::ast::Item>` in the current scope
  --> /Users/nabeelomer/.cargo/registry/src/github.com-1ecc6299db9ec823/pear_codegen-0.0.10/src/lib.rs:72:45
   |
72 |             let mut new_item = item.clone().unwrap();
   |                                             ^^^^^^

   Compiling lru-cache v0.1.1
error[E0599]: no method named `unwrap` found for type `syntax::ptr::P<syntax::ast::Expr>` in the current scope
   --> /Users/nabeelomer/.cargo/registry/src/github.com-1ecc6299db9ec823/pear_codegen-0.0.10/src/lib.rs:174:40
    |
174 |     let mut param_expr = param.clone().unwrap();
    |                                        ^^^^^^

error[E0599]: no method named `unwrap` found for type `syntax::ptr::P<syntax::ast::Expr>` in the current scope
   --> /Users/nabeelomer/.cargo/registry/src/github.com-1ecc6299db9ec823/pear_codegen-0.0.10/src/lib.rs:228:37
    |
228 |     let mut new_expr = expr.clone().unwrap();
    |                                     ^^^^^^

error[E0599]: no method named `unwrap` found for type `syntax::ptr::P<syntax::ast::Expr>` in the current scope
   --> /Users/nabeelomer/.cargo/registry/src/github.com-1ecc6299db9ec823/pear_codegen-0.0.10/src/lib.rs:250:43
    |
250 |     let mut unwrapped_expr = expr.clone().unwrap();
    |                                           ^^^^^^

error[E0599]: no method named `unwrap` found for type `syntax::ptr::P<syntax::ast::Block>` in the current scope
   --> /Users/nabeelomer/.cargo/registry/src/github.com-1ecc6299db9ec823/pear_codegen-0.0.10/src/lib.rs:345:71
    |
345 |             let new_block = gen_stmt(ecx, input, VecDeque::from(block.unwrap().stmts));
    |                                                                       ^^^^^^

error[E0599]: no method named `unwrap` found for type `syntax::ptr::P<syntax::ast::Block>` in the current scope
   --> /Users/nabeelomer/.cargo/registry/src/github.com-1ecc6299db9ec823/pear_codegen-0.0.10/src/lib.rs:357:76
    |
357 |             let new_block = gen_stmt(ecx, input, VecDeque::from(true_block.unwrap().stmts));
    |                                                                            ^^^^^^

error[E0599]: no method named `unwrap` found for type `syntax::ptr::P<(syntax::codemap::Spanned<syntax::ast::Mac_>, syntax::ast::MacStmtStyle, syntax::util::ThinVec<syntax::ast::Attribute>)>` in the current scope
   --> /Users/nabeelomer/.cargo/registry/src/github.com-1ecc6299db9ec823/pear_codegen-0.0.10/src/lib.rs:429:32
    |
429 |             let mac = mac_stmt.unwrap().0;
    |                                ^^^^^^

error: aborting due to 7 previous errors

error: Could not compile `pear_codegen`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by SergioBenitez** [comment]
This is due to a breaking change in the latest nightly. A new release will be posted as soon as possible that fixes this issue. Until then, please use `nightly-2017-12-16`.

**Comment by nabeelomer** [comment]
@SergioBenitez thanks!

**Comment by SergioBenitez** [comment]
Fixed in Rocket v0.3.5.

**Comment by Asone** [comment]
Hi @SergioBenitez, i don't know if this could be considered as related but it looks like we have a similar issue lately for which i opened an issue [here](https://github.com/rust-lang/rust/issues/57120) so i referenced this issue in the linked issue. 

Any tip would be welcome so we can fix the bug. Thanks a lot !
